File: mobile-client/android/UseCase/System/Security/ApplyRequestSignatureUseCase.kt
package app.mobiling.client.usecase.system.security

import app.mobiling.client.data.system.security.RequestSignatureApplier
import okhttp3.Request

class ApplyRequestSignatureUseCase(private val requestSignatureApplier: RequestSignatureApplier) {
    operator fun invoke(request: Request): Request = requestSignatureApplier.apply(request)
}

File: client/ios/UseCase/System/Security/ApplyRequestSignatureUseCase.swift
import Foundation

public struct ApplyRequestSignatureUseCase {
    private let requestSignatureApplier: RequestSignatureApplier

    public init(requestSignatureApplier: RequestSignatureApplier) {
        self.requestSignatureApplier = requestSignatureApplier
    }

    public func callAsFunction(request: URLRequest) -> URLRequest {
        requestSignatureApplier.apply(to: request)
    }
}
